### PR TITLE
Allow access to NavigationMapboxMap and MapboxNavigation

### LINKED ISSUE #1177: NavigationView - Allow Hide/Remove NavigationMapRoute
I'm currently using `NavigationView` with `MapMatching`, the route which is overlayed onto the map seems to cut allot of corners, so I have plotted a `Polyline` over the top of the map.

I would like to hide/remove the `NavigationMapRoute` i can see `navigationView.drawRoute()` however, I don't believe `navigationView.removeRoute()` is available.

Is their anyway to hide this or if i could make the colour of the line `Transparent` (this will allow me to keep the route turn arrows on the map.

### PR DESCRIPTION
**Pull request by danesfeder**
Closes #1177 

Add APIs to the `NavigationView` that give access to the `NavigationMapboxMap` and `MapboxNavigation` instances once the view has been initialized / navigation has started.  

This PR also aims to add meaningful instrumentation tests, that can run quickly and verify different parts of the `NavigationView`.  Two were added to test the new APIs added, as well as an orientation test to make sure we don't break anything on rotation events.

### CODE REVIEW

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 1]
```diff
@@ -381,16 +381,27 @@ public void initialize(OnNavigationReadyCallback onNavigationReadyCallback) {
...
+
+  /**
+   * Returns the instance of {@link MapboxNavigation} powering the {@link NavigationView}
+   * once navigation has started.  Will return null if navgiation has not been started with
+   * {@link NavigationView#startNavigation(NavigationViewOptions)}.
+   *
+   * @return mapbox navigation, or null if navigation has not started
+   */
+  @Nullable
+  public MapboxNavigation retriveMapboxNavigation() {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Typo - `*retrive`

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java [hunk 2]
```diff
@@ -381,16 +381,27 @@ public void initialize(OnNavigationReadyCallback onNavigationReadyCallback) {
...
-   * supported. This returns null until the view is initialized
+   * supported. This returns null until the view is initialized.
+   * <p>
+   * The {@link NavigationMapboxMap} gives direct access to the map UI (location marker, route, etc.).
    *
-   * @return mapbox map object, or null if view has not been initialized
+   * @return navigation mapbox map object, or null if view has not been initialized
    */
   @Nullable
-  public MapboxMap retrieveMapboxMap() {
```
Comment (Guardiola31337) [inline libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Are we ok here breaking SemVer?
If so, we should document this in the release notes.

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java (hunk 1 above)
Comment (danesfeder) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: @Guardiola31337 great catch, thanks!

File: libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java (hunk 2 above)
Comment (danesfeder) [reply libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java]: Yep, will update release notes / documentation

File: circle.yml [hunk 3]
```diff
@@ -55,7 +59,21 @@ jobs:
...
+              --app app/build/outputs/apk/debug/app-debug.apk \
+              --device-ids shamu \
+              --os-version-ids 22 \
+              --locales en \
+              --orientations portrait \
+              --timeout 5m
+            gcloud firebase test android run --type instrumentation \
+              --app app/build/outputs/apk/debug/app-debug.apk \
+              --test app/build/outputs/apk/androidTest/debug/app-debug-androidTest.apk \
+              --device model=hammerhead,version=21,locale=en,orientation=portrait  \
```
Comment (Guardiola31337) [inline circle.yml]: What about setting the locale to German instead of English? We're already testing `en` 👇

File: circle.yml [hunk 4]
```diff
@@ -55,7 +59,21 @@ jobs:
...
+              --device model=athene,version=23,locale=fr,orientation=landscape \
+              --device model=sailfish,version=26,locale=es,orientation=portrait \
+              --device model=walleye,version=28,locale=en,orientation=portrait \
```
Comment (Guardiola31337) [inline circle.yml]: What about setting the orientation to landscape here? 2 portrait / 2 landscape

**Comment by danesfeder** [comment]
@Guardiola31337 this is updated and ready for another round of 👀

File: app/src/androidTest/java/testapp/activity/BaseNavigationActivityTest.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+  @Before
+  public void beforeTest() {
+    try {
+      idlingResource = new OnNavigationReadyIdlingResource(rule.getActivity());
+      IdlingRegistry.getInstance().register(idlingResource);
+      checkViewIsDisplayed(R.id.navigationView);
+      navigationView = idlingResource.getNavigationView();
+    } catch (IdlingResourceTimeoutException idlingResourceTimeoutException) {
+      Timber.e("Idling resource timed out. Couldn't not validate if navigation is ready.");
```
Comment (Guardiola31337) [inline app/src/androidTest/java/testapp/activity/BaseNavigationActivityTest.java]: Typo `*Couldn't not`

File: app/src/androidTest/java/testapp/NavigationViewOrientationTest.java [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  public void onOrientationPortrait_navigationContinuesRunning() {
+    if (checkOrientation(Configuration.ORIENTATION_PORTRAIT)) {
+      return;
+    }
+    validateTestSetup();
+
+    changeOrientation(orientationPortrait());
+  }
+
+  // TODO create test rule for this to conditionally ignore
```
Comment (Guardiola31337) [inline app/src/androidTest/java/testapp/NavigationViewOrientationTest.java]: Let's revisit `Assumptions and Conditional Test Execution` in a future PR.
For reference 👀 https://www.codeaffine.com/2013/11/18/a-junit-rule-to-conditionally-ignore-tests/

File: circle.yml [hunk 7]
```diff
@@ -36,10 +36,14 @@ jobs:
 #          name: Sonarcloud upload
 #          command: make sonarqube
       - run:
-          name: Build testapp APK
+          name: Build Test Application APK
           command: |
             ./gradlew accessToken
             ./gradlew app:assembleDebug
+      - run:
```
Comment (Guardiola31337) [inline circle.yml]: I guess we want this step to be after `Build release to test ProGuard rules` one. No need to run it if `Build release to test ProGuard rules` fails right?

**Comment by Guardiola31337** [comment]
```
java.lang.NullPointerException: Attempt to invoke virtual method 'com.mapbox.mapboxsdk.camera.CameraPosition com.mapbox.mapboxsdk.maps.MapboxMap.getCameraForLatLngBounds(com.mapbox.mapboxsdk.geometry.LatLngBounds, int[])' on a null object reference
at com.mapbox.services.android.navigation.ui.v5.camera.DynamicCamera.createCameraPosition(DynamicCamera.java:151)
at com.mapbox.services.android.navigation.ui.v5.camera.DynamicCamera.createZoom(DynamicCamera.java:112)
at com.mapbox.services.android.navigation.ui.v5.camera.DynamicCamera.zoom(DynamicCamera.java:64)
at com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera.animateCameraFromLocation(NavigationCamera.java:353)
at com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera.access$400(NavigationCamera.java:36)
at com.mapbox.services.android.navigation.ui.v5.camera.NavigationCamera$1.onProgressChange(NavigationCamera.java:53)
at com.mapbox.services.android.navigation.v5.navigation.NavigationEventDispatcher.onProgressChange(NavigationEventDispatcher.java:144)
at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorThreadListener.onNewRouteProgress(RouteProcessorThreadListener.java:34)
at com.mapbox.services.android.navigation.v5.navigation.RouteProcessorHandlerCallback$1.run(RouteProcessorHandlerCallback.java:97)
at android.os.Handler.handleCallback(Handler.java:789)
at android.os.Handler.dispatchMessage(Handler.java:98)
at android.os.Looper.loop(Looper.java:164)
at android.app.ActivityThread.main(ActivityThread.java:6541)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

First crash found as part of the Instrumentation Tests 👏:clap:

**Comment by Danny-James** [comment]
@danesfeder what is the process for hiding the NavigationMapRoute, I don't want to just call `removeRoute()` as I still want to see the `Turn Arrows`, is their a documented way on hiding the Navigation Line?

**Pull request merged**